pgoapi/pgoapi.py

```diff
@@ -439,31 +441,39 @@ def cleanup_pokemon(self, inventory_items=None):
             if len(pokemons) > self.MIN_SIMILAR_POKEMON:
                 # highest CP pokemon first
                 pokemons = sorted(pokemons, key=lambda pok: (pok.cp, pok.iv), reverse=True)
-                pokemons_keep_counter = 0
-                pokemons_keep_iv_counter = 0
+                self._pokemons_keep_counter = 0
+                self._pokemons_keep_iv_counter = 0
 
                 for pokemon in pokemons:
-                    # never release favorites and other defined pokemons
-                    if pokemon.is_favorite or pokemon.pokemon_id in self.keep_pokemon_ids:
-                        pokemons_keep_counter += 1
-                        continue
-                    # release defined throwaway pokemons
-                    elif pokemon.pokemon_id in self.throw_pokemon_ids:
-                        self.do_release_pokemon(pokemon)
-                    # keep high-iv pokemons based on config values
-                    elif pokemon.iv > self.KEEP_IV_OVER and pokemons_keep_iv_counter < self.MAX_POKEMON_HIGH_IV and \
-                                    pokemon.cp > (pokemons[0].cp * self.KEEP_IV_MIN_PERCENT_CP / 100):
-                        pokemons_keep_iv_counter += 1
-                        pokemons_keep_counter += 1
-                        continue
-                    # keep high-cp pokemons and first MIN_SIMILAR_POKEMON amount of pokemons
-                    elif pokemon.cp > self.KEEP_CP_OVER or pokemons_keep_counter < self.MIN_SIMILAR_POKEMON:
-                        pokemons_keep_counter += 1
-                        continue
-                    # release all other pokemons
-                    else:
+                    if self.is_pokemon_eligible_for_transfer(pokemon, pokemons[0]):
                         self.do_release_pokemon(pokemon)
 
+                # reset the counters just in case
+                self._pokemons_keep_iv_counter = 0
+                self._pokemons_keep_counter = 0
+
+    def is_pokemon_eligible_for_transfer(self, pokemon, best_pokemon=None):
```

Comment:
(True, True) = Release and do not increment :)
(False, False) = Do not release and increment both
(False,True) = Do not release increment 1 counter
(True, False) = Do not release and increment other counter 
:) 

> On Jul 26, 2016, at 1:39 PM, Marcel <notifications@github.com> wrote:
> 
> In pgoapi/pgoapi.py <https://github.com/j-e-k/pgoapi/pull/110#discussion_r72299445>:
> 
> >                          self.do_release_pokemon(pokemon)
> >  
> > +                # reset the counters just in case
> > +                self._pokemons_keep_iv_counter = 0
> > +                self._pokemons_keep_counter = 0
> > +
> > +    def is_pokemon_eligible_for_transfer(self, pokemon, best_pokemon=None):
> Beware if you want to remove the attributes from the class, is_pokemon_eligible_for_transfer will need 3 different return values, only True / False won't work.
> 1. Release pokemon
> 2. Keep and increase keep_counter
> 3. Keep and increase keep_counter AND keep_iv_counter
> 
> thanks for merging btw :)
> 
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub <https://github.com/j-e-k/pgoapi/pull/110/files/059a8faaaf352b20f26bd19ade6dd83c0b7d0274..892d52f2f3e1ab46288dce89ad7b80991d14387b#r72299445>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AF92WG7QEz-BzUpb6hsu7XgKzSWDEm2Nks5qZkY_gaJpZM4JVKF->.
>